core/src/utils/content/index.ts

```diff
@@ -0,0 +1,130 @@
+    return true;
+  }
+  return false;
+}
+
+const getScrollContainer = (node: Element): HTMLElement | null => {
+  if (hasOverflowScroll(node)) {
+    return node as HTMLElement;
+  }
+  for (const item of Array.from(node.children)) {
+    if (item && hasOverflowScroll(item)) {
+      return item as HTMLElement;
+    }
+  }
+  return null;
+}
+
+/**
+ * Waits for the element host fully initialize before
+ * returning the inner scroll element.
+ *
+ * For `ion-content` the scroll target will be the result
+ * of the `getScrollElement` function.
+ *
+ * For custom implementations it will be the element host
+ * or a selector within the host, if supplied through `scrollTarget`.
+ */
+export const getScrollElement = async (el: Element) => {
+  await new Promise(resolve => componentOnReady(el, resolve));
+
+  if (el?.tagName.toLowerCase() === ION_CONTENT_ELEMENT_SELECTOR) {
+    return (el as HTMLIonContentElement).getScrollElement();
+  }
+
+  const scrollContainer = getScrollContainer(el);
+
+  if (scrollContainer) {
+    return scrollContainer;
+  }
+
+  return el as HTMLElement;
+}
+
+/**
+ * Queries the element matching the selector for IonContent.
+ * See ION_CONTENT_SELECTOR for the selector used.
+ */
+export const findIonContent = (el: Element) => {
+  /**
+   * First we try to query the custom scroll host selector in cases where
+   * the implementation is using an outer `ion-content` with an inner custom
+   * scroll container.
+   */
+  const customContentHost = el.querySelector<HTMLElement>(ION_CONTENT_CLASS_SELECTOR);
+  if (customContentHost) {
+    return customContentHost;
+  }
+  return el.querySelector<HTMLElement>(ION_CONTENT_SELECTOR);
+}
+
+/**
+ * Queries the closest element matching the selector for IonContent.
+ */
+export const findClosestIonContent = (el: Element) => {
+  return el.closest<HTMLElement>(ION_CONTENT_SELECTOR);
+}
+
+/**
+ * Scrolls to the top of the element. If an `ion-content` is found, it will scroll
+ * using the public API `scrollToTop` with a duration.
+ */
+export const scrollToTop = (el: HTMLElement, durationMs: number): Promise<any> => {
+  if (el.tagName.toLowerCase() === ION_CONTENT_ELEMENT_SELECTOR) {
+    const content = el as HTMLIonContentElement;
+    return content.scrollToTop(durationMs);
+  }
+  return Promise.resolve(el.scrollTo({
+    top: 0,
+    left: 0,
+    behavior: durationMs > 0 ? 'smooth' : 'auto'
+  }));
+}
+
+/**
+ * Scrolls to a specific x/y location. If an `ion-content` is found, it will scroll
+ * using the public API `scrollToPoint` with a duration.
+ */
+export const scrollByPoint = (el: HTMLElement, x: number, y: number, durationMs: number) => {
+  if (el.tagName.toLowerCase() === ION_CONTENT_ELEMENT_SELECTOR) {
+    const content = el as HTMLIonContentElement;
+    return content.scrollByPoint(x, y, durationMs);
+  }
+  return Promise.resolve(el.scrollTo({
+    top: y,
+    left: x,
+    behavior: durationMs > 0 ? 'smooth' : 'auto'
+  }));
```

Comment:
The comment references `scrollToPoint`, and the use of `scrollTo` would mirror that behavior, but the actual function is called and uses `scrollByPoint`, which is slightly different. (Check their implementations in `ion-content`.) It looks like `scrollByPoint` was your intention here since it's used in the `scroll-assist` shim, so the comment and fallback should be updated.

Reply:
Very good catch! Updated the comment and the fallback implementation for the `HTMLElement`, to use scroll by point behavior.